Test/test_installation/CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+#   - No XML headers etc. are included in the import headers
+#   - That libraries are present and functions can be called
+project(TestInstallation)
+
+set_property(GLOBAL PROPERTY C_STANDARD          99)
+set_property(GLOBAL PROPERTY C_STANDARD_REQUIRED ON)
+
+set(FMIL_HOME "unset" CACHE FILEPATH "Path to FMIL installation")
+
+message(STATUS "FMIL_HOME: ${FMIL_HOME}")
+if (FMIL_HOME STREQUAL "unset" OR FMIL_HOME STREQUAL "")
+    message(FATAL_ERROR "Unset config variable: FMIL_HOME")
+endif()
+
+# Macro that sets MSVC RTL for the target to the same one as FMIL was compiled with.
+macro(target_set_msvcrt TARGET_NAME)
+    if(FMILIB_BUILD_WITH_STATIC_RTLIB)
+        set_property(TARGET ${TARGET_NAME} PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded")
+    else()
+        set_property(TARGET ${TARGET_NAME} PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
+    endif()
+endmacro()
+
+enable_testing()
+
+# ==============================================================================
+# Test static installation
+# ==============================================================================
+
+# Import FMIL
+add_library(fmilib STATIC IMPORTED)
+set_property(TARGET fmilib PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${FMIL_HOME}/include")
+if(MSVC)
+    set_property(TARGET fmilib PROPERTY IMPORTED_LOCATION "${FMIL_HOME}/lib/fmilib.lib")
+else()
+    set_property(TARGET fmilib PROPERTY IMPORTED_LOCATION "${FMIL_HOME}/lib/libfmilib.a")
+endif()
+
+# Build the test
+add_executable(main main.c)
+target_link_libraries(main PRIVATE fmilib)
+if(WIN32)
+    target_link_libraries(main PRIVATE fmilib Shlwapi)
+else()
+    target_link_libraries(main PRIVATE fmilib dl)
+endif()
+if(WIN32)
+    # Needed for compiling against static libs. FIXME: Should not be needed.
+    target_compile_definitions(main PRIVATE FMILIB_STATIC_LIB_ONLY)
+endif()
+if(MSVC)
+    target_set_msvcrt(main)
+endif()
+
+add_test(test_installation main)
+
+# ==============================================================================
+# Test shared installation
+# ==============================================================================
+
+# Import FMIL
+add_library(fmilib_shared SHARED IMPORTED)
+set_property(TARGET fmilib_shared PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${FMIL_HOME}/include")
+if(MSVC)
+    set_property(TARGET fmilib_shared PROPERTY IMPORTED_LOCATION "${FMIL_HOME}/lib/fmilib_shared.dll")
+    set_property(TARGET fmilib_shared PROPERTY IMPORTED_IMPLIB   "${FMIL_HOME}/lib/fmilib_shared.lib")
+elseif(MINGW)
+    set_property(TARGET fmilib_shared PROPERTY IMPORTED_LOCATION "${FMIL_HOME}/lib/libfmilib_shared.dll")
+    set_property(TARGET fmilib_shared PROPERTY IMPORTED_IMPLIB   "${FMIL_HOME}/lib/libfmilib_shared.dll.a")
+else()
+    set_property(TARGET fmilib_shared PROPERTY IMPORTED_LOCATION "${FMIL_HOME}/lib/libfmilib_shared.so")
+endif()
+
+# Build the test
+add_executable(main_shared main.c)
+target_link_libraries(main_shared PRIVATE fmilib_shared)
```

Comment:
Isn't this line covered by both of the branches in the following if?

Reply:
You're right, and same for static fmilib, will fix